README.md

```diff
@@ -2,43 +2,176 @@
+You are free to set any name you want in the `sandbox-name` parameter, as long as it contains only letters, numbers, underscores and hyphens.
 
 VTEX Toolbelt will upload all your app files to the sandbox specified and will be watching for any changes you make to them.
 
-## Requirements
+There's other forms of use too, if you use the options:
+
+Option|Alias
+---|---
+`vtex watch --webpack <sandbox-name>`|`vtex watch -w <sandbox-name>`
+`vtex watch --server <sandbox-name>`|`vtex watch -s <sandbox-name>`
+
+## Webpack
+
+You can run the VTEX Toolbelt watcher in parallel with the [Webpack](http://webpack.github.io/) watcher if you use the `--webpack` option under the `watch` command.
+
+Make sure that you have a well configured and working [webpack.config.js](http://webpack.github.io/docs/tutorials/getting-started/#config-file) on the root of your project.
+
+```sh
+vtex watch --webpack <sandbox-name>
+```
+
+## Webpack Dev Server
+
+You can also run the VTEX Toolbelt watcher in parallel with the [Webpack Dev Server](http://webpack.github.io/docs/webpack-dev-server.html) watcher if you use the `--server` option under the `watch` command.
+
+As Webpack Dev Server uses Webpack, you also need a webpack.config.js file on the root of yout project.
+
+```sh
+vtex watch --server <sandbox-name>
+```
+
+You need to call it this way if you want to enable [Hot Module Replacement](http://webpack.github.io/docs/hot-module-replacement-with-webpack.html), see below for more information on how to configure your project and OS in order to make it work properly.
+
+# Livereload
+
+Add to your layout the following script:
+
+```html
+  <script src="http://localhost:35729/livereload.js?snipver=1"></script>
+```
+
+# Hot Module Replacement
+
+In order to make the Hot Module Replacement work, besides running the `watch` command with the `--server` option, you need to:
+
+## Edit your hosts file
+
+The hosts file is a computer file used by an operating system to map hostnames to IP addresses. 
+
+On Linux and Mac you can find the file on `/etc/hosts`, you need `sudo` to be able to edit this file.
+On Windows you can find it on `C:\Windows\System32\Drivers\etc\hosts`, you may need administrative privileges to edit the file.
+
+Append to the end of your file the following:
+
+```plain
+127.0.0.1      <name-of-your-store>.beta.myvtex.com
+```
+
+127.0.0.1 is your localhost. It may vary on different machines.
+
+## Add this snippet to your Webpack entry point
+
+```javascript
+// Enable react hot loading with external React
+if (module.hot) {
+  window.RootInstanceProvider = require('react-hot-loader/Injection').RootInstanceProvider;
+```
+
+For more on that matter, see [this](https://github.com/gaearon/react-hot-loader/tree/master/docs#usage-with-external-react).
+
+## Configure your Webpack config file
+
+You need to add to your webpack.config.js file the options that Webpack Dev Server uses in order to work, it can be something like:
```

Comment:
Maybe show this snippet instead:
https://github.com/vtex-apps/storefront-theme/blob/master/webpack.config.js#L121-L138